sympy/parsing/C/parser.py

```diff
@@ -0,0 +1,239 @@
+
+    def _output_module(self, mod, out):
+        out.write('===== %s.py ==================================================\n' % mod.full_name)
+        mod.output(Writer(out))
+        for submodule in mod.submodules.values():
+            self._output_module(submodule, out)
+
+    def output_all(self, out):
+        self._output_module(self.root_module, out)
+
+    def parse(self, filenames, flags):
+        abs_filenames = [os.path.abspath(f) for f in filenames]
+        self.filenames.update(abs_filenames)
+
+        for filename in abs_filenames:
+            if os.path.splitext(filename)[1] != '.h':
+                self.tu = self.index.parse(
+                    filename,
+                    args=flags,
+                    options=TranslationUnit.PARSE_DETAILED_PROCESSING_RECORD
+                )
+                self.transform(self.tu.cursor, self.root_module)
+
+    def parse_text(self, content, flags):
+        for f, c in content:
+            abs_filename = os.path.abspath(f)
+            self.filenames.add(abs_filename)
+
+            self.tu = self.index.parse(
+                f,
+                args=flags,
+                unsaved_files=[(f, c)],
+                options=TranslationUnit.PARSE_DETAILED_PROCESSING_RECORD
+            )
+            self.transform(self.tu.cursor, self.root_module)
+
+    def transform(self, node, context = None):
+
+
+    def transform_function_decl(self, node, context):
+        function = Function(context, node.spelling)
+        try:
+            children = node.get_children()
+            child = next(children)
+
+            # If the node has any children, the first children will be the
+            # return type and namespace for the function declaration. These
+            # nodes can be ignored.
+            # print("FIRST CHILD", child.kind)
+            while child.kind == CursorKind.NAMESPACE_REF:
+                child = next(children)
+                # print("NS CHILD", child.kind)
+            while child.kind == CursorKind.TYPE_REF:
+                child = next(children)
+                # print("TYPE REF CHILD", child.kind)
+
+            # Subsequent nodes will be the parameters for the function.
+            try:
+                while True:
+                    decl = self.handle(child, function)
+                    if decl:
+                        decl.add_to_context(function)
+                    child = next(children)
+            except StopIteration:
+                pass
+
+        except StopIteration:
+            pass
+
+        # Only return a node if we get function definition. The prototype
+        # can be ignored.
+        if function.statements is not None:
+            return function
+
+     def transform_var_decl(self, node, context):
+        try:
+            children = node.get_children()
+            prev_child = None
+            child = next(children)
+            # print("FIRST CHILD", child.kind)
+
+def convert_file(src_in, src_out):
```

Comment:
If this is a command line tool, it should have 

```py
if __name__ == '__main__':
    main()
```

at the bottom so it can be called like `python -m sympy.parsing.c.parser`